### DISCUSSION TITLE
rpi camera stream: insert date and time in the video stream

### DISCUSSION THREAD

**Discussion by vongillus** [bluenviron/mediamtx#1207] [Questions]
I just succeeded in configuring a rpi with its camera with both RTSP and HLS low latency. Thanks a lot for your work on this software!

I was wondering, is there any way to insert the date and time on the video? It will be a way to check that the video is not frozen. I mean in a more general way, to insert text on the video.

**Comment by aler9** [comment]
Hello, in order to insert text on video you generally have to decode and re-encode the stream, you can do it with GStreamer but it also consumes a lot of CPU.

Regarding the Raspberry Pi Camera, it may be possible possible to do that without consuming CPU since we have access to raw frames before they're encoded into H264, but it's currently not implemented.

libcamera-vid is able to add date and time to the video by using the post-process stage `annotate_cv_stage`:
https://www.raspberrypi.com/documentation/computers/camera_software.html#annotate_cv-stage

I've created a feature request  in order to track any future development inside the server: #1288
